src/utils/mod.rs

```diff
@@ -1,49 +0,0 @@
-pub(crate) mod random;
```

Comment:
This is actually better. What is the reason for moving this outside to utils.rs? Utils module will contain many submodules like random in future.

Reply:
It's still there, just using the new naming convention. Was for consistency (the same happens with rdd, io and other modules in the future) eg:
```
rdd.rs
rdd/co_grouped_rdd.rs
rdd/...
...
utils.rs
utils/random.rs // <-- this could be expanded to utils/random/other_submod.rs in the future.
utils/...
```
I have no strong opinion on using either of the naming conventions (eg. using utils/mod.rs instead of utils.rs + utils/**), AFAIK functionally are the same; but IMO we should stick to one only.

I can change it back to utils/mod.rs (and rdd/mod.rs etc.) if you prefer, although it's usually more clear when you use the name directly instead of having different "mod.rs" open in the IDE. But as said, no strong opinion.

Reply:
I feel that utils/mod.rs and rdd/mod.rs is better and keep everything contained in single folder. I noticed another minor thing, utils.rs seems to be a completely new file instead of a moved file. Won't it mess with git file history?